front/app/modules/commercial/analytics/admin/hooks/useActiveUsers/query.ts

```diff
@@ -0,0 +1,93 @@
+// utils
+import {
+  getProjectFilter,
+  getDateFilter,
+  getDateFilterLastPeriod,
+  getInterval,
+} from '../../utils/query';
+
+// typings
+import { Query, QuerySchema } from '../../services/analyticsFacts';
+import { QueryParameters } from './typings';
+
+export const query = ({
+  projectId,
+  startAtMoment,
+  endAtMoment,
+  resolution,
+}: QueryParameters): Query => {
+  const timeSeriesQuery: QuerySchema = {
+    fact: 'participation',
+    filters: {
+      'dimension_user.role': ['citizen', null],
+      ...getProjectFilter('dimension_project', projectId),
+      ...getDateFilter('dimension_date_created', startAtMoment, endAtMoment),
+    },
+    groups: `dimension_date_created.${getInterval(resolution)}`,
+    aggregations: {
+      dimension_user_id: 'count',
+    },
+  };
+
+  const activeUsersWholePeriodQuery: QuerySchema = {
+    fact: 'participation',
+    filters: {
+      'dimension_user.role': ['citizen', null],
+      ...getProjectFilter('dimension_project', projectId),
+      ...getDateFilter('dimension_date_created', startAtMoment, endAtMoment),
+    },
+    aggregations: {
+      dimension_user_id: 'count',
+    },
+  };
+
+  const activeUsersLastPeriodQuery: QuerySchema = {
+    fact: 'participation',
+    filters: {
+      'dimension_user.role': ['citizen', null],
```

Comment:
do we need the `null` here? aren't users registered and with the citizen role? same above

Reply:
Yeah James also said that. I guess it doesn't matter for the response right? I left it now for consistency with I am also happy to remove it this is confusing

Reply:
ok yeah, it only make sense in visits. on active users or participations or basically everywhere else we would just look for citizens, wouldn't make a difference on the response though.

Reply:
I'm not sure if I understand the role part completely... where should I change this filter and how should I change it? Or shall I just leave it like this?

Reply:
yeah, maybe its not clear that anonymous visits have a user role in the analytics data schema.. but it was made to support visits imported from matomo.
So if you want to exclude admin participations for example you would use `'dimension_user.role': 'citizen'`
if you remove the whole filter you would get participations for every type of registered user.